black-jack-redo.rb

```diff
@@ -25,88 +25,205 @@ def print_with_stars(string)
+		card_suit = "Clubs\n"
+	else 
+		puts 'Something went wrong!'
 	end
-	return count4, string
+
+	return card_suit
+end
+
+def get_card(card)
+	user_card = ''
+	if (card[1] == 'A') || (card[1] == 'a')
+		user_card = 'Ace of ' + get_suit(card)
+	elsif card[1] == 'J'
+		user_card = 'Jack of ' + get_suit(card)
+	elsif card[1] == 'Q'
+		user_card = 'Queen of ' + get_suit(card)
+	elsif card[1] == 'K'
+		user_card =  'King of ' + get_suit(card)
+	elsif (card[1].to_i >= 2) || (card[1..-1].to_i <= 10)
+		user_card = "#{card[1..-1]} of " + get_suit(card)
+	else
+		puts 'Something went wrong when retrieving the card!'
+		return
+	end
+	return user_card
 end
 
-def handcount(array)
-	count3 = 0
-	count4 = 0
-	while count3 < array.length
-		if array[count3][:dealt] == true
-			count4 += array[count3][:value]
+def viewhand(array, is_user)
+	count = 0
+	hand_value = 0
+	if is_user == 'true'
+		user_hand = "You have #{array.length} cards in your hand! " + "\n" + "Your hand includes the following:" + "\n"
+	else
+		user_hand = "The dealer has:" + "\n"
+	end
+
+	while count < array.length
+		string = array[count]
+		if string[1] == 'A'
+			user_hand = user_hand + 'Ace of ' + get_suit(array[count])
+			hand_value += 11
+			count += 1
+		elsif string[1] == 'a'
+			user_hand = user_hand + 'Ace of ' + get_suit(array[count])
+			hand_value += 1
+			count += 1
+		elsif string[1] == 'J'
+			user_hand = user_hand + 'Jack of ' + get_suit(array[count])
+			hand_value += 10
+			count += 1
+		elsif string[1] == 'Q'
+			user_hand = user_hand + 'Queen of ' + get_suit(array[count])
+			hand_value += 10
+			count += 1
+		elsif string[1] == 'K'
+			user_hand = user_hand + 'King of ' + get_suit(array[count])
+			hand_value += 10
+			count += 1
+		elsif (string[1].to_i >= 2) || (string[1..-1].to_i <= 10)
+			user_hand = user_hand + "#{string[1..-1]} of " + get_suit(array[count])
+			hand_value += string[1..-1].to_i
+			count += 1
 		else
+			puts 'Something went wrong!'
+			return
 		end
-		count3 += 1
 	end
-	return count4
-end
 
-def dealer(count)
-	number = rand(4)
-	number = 21 - number
-	if count == number
-		return "You and the dealer both had #{number}, it's a tie!"
-	elsif count < number
-		return "The dealer had #{number} and you had #{count}, you lose!"
-	else count > number
-		return "The dealer had #{number} and you had #{count}, you win!"
+	if is_user == 'true'
+		user_hand = user_hand + "Your cards add up to #{hand_value} points!" + "\n"
+	else
+		user_hand = user_hand + "The dealers cards add up to #{hand_value} points!" + "\n"
 	end
+	return user_hand, hand_value
 end
 
-def draw(array)
-	number = array.length - 1
-	puts array[number]
-	return array[number]
+def get_value_of_hand(current_hand)
+	user_hand, hand_value = viewhand(current_hand, 'true')
+	return hand_value
 end
 
-def drawone(array, cards, count)
-	bust = false
-	number = rand(52)
-	if array[number][:dealt] == true
-		draw(array, cards)
+def who_wins(array, current_hand, dealer_hand)
+	string = ''
```

Comment:
You don't need this variable declaration. Your `print_with_stars` method will take a string by itself - you don't need to append something to this one.